db/mongodb/basic/driver.rkt

```diff
@@ -177,18 +177,20 @@
 (provide/contract
  [mongo-db-collections (mongo-db? . -> . (listof string?))])
 (define (mongo-db-collections db)
-  (define name (mongo-db-name db))
-  (define ans (mongo-find (mongo-db-mongo db) (format "~a.system.namespaces" name) empty))
-  (define name-rx (regexp (format "^~a\\.(.+)$" (regexp-quote name))))
-  (for/fold ([l empty])
-    ([c ans])
-    (define n (hash-ref c 'name))
-    (match (regexp-match name-rx n)
-      [(list _ name)
-       (if (regexp-match #rx"\\$" name)
-           l
-           (list* name l))]
-      [#f l])))
+  (let ([collections (mongo-db-collections db (list
```

Comment:
Is this a typo? Does it call itself?

Reply:
Should it be `mongo-db-execute-command!`?

Reply:
Yes, that is right. I did not notice how I made a mistake. In the current version of my branch, the master fixed it to a correct name.